src/schedule/ll_schedule.c

```diff
@@ -167,19 +167,24 @@ static void schedule_ll_client_reschedule(struct ll_schedule_data *sch)
 		list_for_item(tlist, &sch->tasks) {
 			task = container_of(tlist, struct task, list);
 
+			/* only check tasks asked for rescheduling */
+			if (task->state != SOF_TASK_STATE_RESCHEDULE)
+				continue;
+
 			/* update to use the earlier tick */
 			if (task->start < next_tick) {
 				next_tick = task->start;
-				task_take_dbg = task;
+				task_take = task;
```

Comment:
since we have `next_tick` above - wouldn't it make sense to name the respective task `next_task`?

Reply:
> since we have `next_tick` above - wouldn't it make sense to name the respective task `next_task`?

um, the "take" was named here to indicate that the specific task is taken into account, but the task itself is not necessarily the one to be scheduled next, that's what I want to keep as is.